services/DynamoService.test.ts

```diff
@@ -85,36 +82,45 @@ describe("DynamoService", () => {
   });
 
   describe('putItem', () => {
-    const item = { commonName: 'Treebeard', sidarinName: 'Fangorn', entishName: 'Not stored due to buffer overflow', orcishName: '' };
+    // We are marshalling the item as that is what we expect to receive from the User model
+    const item = DynamoDB.Converter.marshall(
+      {
+        commonName: 'Treebeard',
+        entishName: 'Not stored due to buffer overflow',
+        orcishName: '',
+        sidarinName: 'Fangorn',
+      },
+      { convertEmptyValues: true },
+    );
     const tableName = 'Ents';
 
     it('puts to a DynamoDB table', async () => {
       await service.putItem(item, tableName);
-      expect(mockPut).toHaveBeenCalledWith({ Item: item, TableName: tableName }, expect.any(Function));
+      expect(mockPut).toHaveBeenCalledWith(
+        { Item: item, TableName: tableName },
+        expect.any(Function),
+      );
     });
 
     // The DynamoDB API breaks if empty strings are passed in
     it('converts empty strings in user model to nulls', async () => {
       await service.putItem(item, tableName);
-      expect(mockPut.mock.calls[0][0]['Item']['orcishName']).toEqual(null);
+      expect(mockPut.mock.calls[0][0].Item.orcishName).toEqual({ NULL: true });
     });
 
     it('responds to an error with a rejection', async () => {
-      //Fail the test if the expectation in the catch is never reached.
+      // Fail the test if the expectation in the catch is never reached.
       expect.assertions(1);
       const err = new Error('Never is too long a word even for me . . . ') as AWSError;
-      mockPut.mockImplementationOnce((
-        _params,
-        cb?: (err: AWSError, _data) => void,
-      ) => {
-        setTimeout(() => cb && cb(err, undefined), 5);
-        return ({} as AWSRequest<PutItemOutput, AWSError>);
+      mockPut.mockImplementationOnce((_params, cb?: (err: AWSError, _data) => void) => {
+        setTimeout(() => cb?.(err, undefined), 5);
+        return {} as AWSRequest<PutItemOutput, AWSError>;
       });
 
       try {
         await service.putItem(item, tableName);
         throw new Error('you shall not pass! (no, really, this is the wrong error)');
-      } catch (putError) {
+      } catch (putError: unknown) {
```

Comment:
Can we use AWSError here since we know `putError` should equal `err`?
```suggestion
      } catch (putError: AWSError) {
```